src/lib.rs

```diff
@@ -361,6 +361,17 @@ impl<T: PartialEq> PartialEq for CircularQueue<T> {
 
 impl<T: Eq> Eq for CircularQueue<T> {}
 
+/// Convert the CircularQueue into a `Vec<T>` going from the most recently pushed items to the oldest ones.
+impl<T> Into<Vec<T>> for CircularQueue<T> {
+    fn into(mut self) -> Vec<T> {
+        let mut upper_part = self.data.split_off(self.insertion_index);
```

Comment:
This allocates an extra Vec. I would probably expect an `Into<Vec<T>>` to not do any extra allocations. I believe you should be able to do it using swaps and `swap_with_slice()`.